zyte_spider_templates/spiders/ecommerce.py

```diff
@@ -241,8 +256,37 @@ def get_start_request(self, url):
         )
 
     def start_requests(self) -> Iterable[Request]:
-        for url in self.start_urls:
-            yield self.get_start_request(url)
+        if self.args.search_queries:
+            for url in self.start_urls:
+                meta: Dict[str, Any] = {
+                    "crawling_logs": {"page_type": "searchRequestTemplate"},
+                }
+                if self.args.crawl_strategy == EcommerceCrawlStrategy.full:
+                    meta["page_params"] = {"full_domain": get_domain(url)}
+                if self.args.extract_from == ExtractFrom.browserHtml:
+                    meta["zyte_api_automap"] = {"browserHtml": True}
```

Comment:
Do you know if it works properly with AnyResponse dependency? I.e. the switch happens, and also there is no extra raw http request when the browser is used?

Reply:
It most likely did not before [switching this to `zyte_api_provider`](https://github.com/zytedata/zyte-spider-templates/pull/77/commits/9dd4f3828d13065e8caf0d43ad82e567682dd3e3), seems to work as expected [with DynamicDeps](https://github.com/zytedata/zyte-spider-templates/pull/77/commits/cdaea6442c62fcf50374749d2e941b199c187fdf).